dnf/base.py

```diff
@@ -1361,16 +1361,17 @@ def install(self, pkg_spec, reponame=None):
             if not match:
                 raise dnf.exceptions.MarkingError(
                     _('no package matched'), pkg_spec)
+            already_inst = set()
             for sltr in sltrs:
                 if not sltr.matches():
                     continue
                 if reponame is not None:
                     sltr = sltr.set(reponame=reponame)
-                already_inst = self._sltr_matches_installed(sltr)
-                if already_inst:
-                    for package in already_inst:
-                        _msg_installed(package)
+                for pkg in self._sltr_matches_installed(sltr):
```

Comment:
This should be IMO fixed in `get_best_selectors()`. Add this test and make it pass, please:
```
    def test_best_selectors_glob(self):
        subj = dnf.subject.Subject('l*')
        sltrs = subj.get_best_selectors(self.base.sack)
        q = self.base.sack.query().filter(name__glob='l*')
        self.assertEqual(len(sltrs), len(set(map(lambda p: p.name, q))))
```
Every selector should represent one package name